### PR TITLE
Adds module to reassign items to the correct supporters

### PR DESCRIPTION
**Pull request by clarissalimab**
Part of https://github.com/CommitChange/tix/issues/3615

Contains code from https://github.com/CommitChange/houdini/pull/229

This PR has the same code we used on staging to test with the nonprofit, but in a module to make it testeable. Additionally, it creates a table to store the reassignments in case they ask us to revert some record.

### CODE REVIEW

**Review by wwahammy — APPROVED** [review]
Nothing seems unreasonable but it's just hard to review this because it makes my head hurt.

File: lib/reassign_supporter_items.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    def self.reassign_journal_entries(correct_supporter, journal_entries_to_items_with_wrong_supporter, etap_import)
+        return if correct_supporter.blank?
+
+        journal_entries_to_items_with_wrong_supporter.each do |journal_entry|
+            reassign_item(correct_supporter, journal_entry.item, etap_import)
+        end
+    end
+
+    def self.reassign_item(correct_supporter, item, etap_import)
+        activity = Activity.find_by(attachment_id: item.id, supporter: item.supporter)
```
Comment (clarissalimab) [inline lib/reassign_supporter_items.rb]:
```suggestion
        activity = item.activities
```

File: db/migrate/20220209203456_create_reassignments.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateReassignments < ActiveRecord::Migration
+  def change
+    create_table :reassignments do |t|
+      t.references :item, polymorphic: true
+      t.references :e_tap_import, index: true, foreign_key: true
```
Comment (clarissalimab) [inline db/migrate/20220209203456_create_reassignments.rb]: Remove that foreign key

File: lib/reassign_supporter_items.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# License: AGPL-3.0-or-later WITH Web-Template-Output-Additional-Permission-3.0-or-later
+module ReassignSupporterItems
+    def self.perform(etap_import)
+        badly_assigned_items = find_badly_assigned_items(etap_import)
+        ActiveRecord::Base.transaction do
+            badly_assigned_items.each do |items|
+                reassign_journal_entries(items[:supp_through_contact], items[:journal_entries_to_items_with_wrong_supporter], etap_import)
+            end
+        end
```
Comment (clarissalimab) [inline lib/reassign_supporter_items.rb]: Move that transaction into the loop

**Review by wwahammy — APPROVED** [review]
It all makes sense to me.

**Review by wwahammy — APPROVED** [review]

**Review by wwahammy — APPROVED** [review]
Looks good to me

**Pull request merged**